pkg/chrootarchive/cmd_linux.go

```diff
@@ -0,0 +1,41 @@
+package chrootarchive
+
+import (
+	"os/exec"
+	"syscall"
+
+	"github.com/docker/docker/pkg/mount"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+func configureSysProc(cmd *exec.Cmd) {
+	cmd.SysProcAttr = &unix.SysProcAttr{
+		Cloneflags: unix.CLONE_NEWNET | unix.CLONE_NEWPID | unix.CLONE_NEWIPC | unix.CLONE_NEWUTS | unix.CLONE_NEWNS,
+		Pdeathsig:  syscall.SIGKILL,
+	}
+}
+
+// As part of setting up this process we created a new mount ns and a new pid ns.
+// This configures the mount ns to take advantage of that to further isolate the process.
+func setupMountNS() error {
+	// Make everything in new ns slave.
+	// Don't use `private` here as this could race where the mountns gets a
+	//   reference to a mount and an unmount from the host does not propagate,
+	//   which could potentially cause transient errors for other operations,
+	//   even though this should be relatively small window here `slave` should
+	//   not cause any problems.
+	if err := mount.MakeRSlave("/"); err != nil {
+		return errors.Wrap(err, "error remounting rootfs as with slave mounts")
+	}
+
+	if err := mount.Unmount("/proc"); err != nil {
```

Comment:
It probably make sense to reverse these two, so the order will be
(1) unmount /proc
(2) make everything rslave

In this case (2) might be a tiny bit faster.

Reply:
If I don't make it rslave first then it may unmount `/proc` on the parent namespace.